libs/wingc/src/jsify.rs

```diff
@@ -655,107 +656,105 @@ impl<'a> JSifier<'a> {
 		}
 	}
 
-	pub fn jsify_struct_properties(
-		&self,
-		fields: &Vec<StructField>,
-		extends: &Vec<UserDefinedType>,
-		ctx: &mut JSifyContext,
-	) -> CodeMaker {
+	pub fn jsify_struct_env_properties(&self, env: &SymbolEnv, ctx: &mut JSifyContext) -> CodeMaker {
 		let mut code = CodeMaker::default();
-
-		// Any parents we need to get their properties
-		for e in extends {
-			code.line(format!(
-				"...require(\"{}\")().jsonSchema().properties,",
-				struct_filename(&e.root.name)
-			))
-		}
-
-		for field in fields {
+		for (field_name, (.., kind)) in env.symbol_map.iter() {
 			code.line(format!(
-				"{}: {{ {} }},",
-				field.name.name,
-				self.jsify_struct_field_to_json_schema_type(&field.member_type.kind, ctx)
+				"{}: {},",
+				field_name,
+				self.jsify_struct_schema_field(&kind.as_variable().unwrap().type_, ctx)
 			));
 		}
+		code
+	}
 
+	pub fn jsify_struct_schema_required_fields(&self, env: &SymbolEnv) -> CodeMaker {
+		let mut code = CodeMaker::default();
+		code.open("required: [");
+		for (field_name, (_stmt_idx, kind)) in env.symbol_map.iter() {
+			if !matches!(*kind.as_variable().unwrap().type_, Type::Optional(_)) {
+				code.line(format!("\"{}\",", field_name));
+			}
+		}
+		code.close("]");
 		code
 	}
```

Comment:
I see there's a lot of `jsify_struct_*` methods, I wonder if it would help to corral them all in a single file like `struct_schema.rs`? It could make it easier to write unit tests for individual functions and stuff